jsonschema/_validators.py

```diff
@@ -70,13 +73,31 @@ def items(validator, items, instance, schema):
     if not validator.is_type(instance, "array"):
         return
 
-    if validator.is_type(items, "array"):
-        for (index, item), subschema in zip(enumerate(instance), items):
+    if validator.is_type(items, "boolean") and 'prefixItems' in schema:
+        if not items:
+            if len(instance) > len(schema['prefixItems']):
+                yield ValidationError(
+                    "%r has more items than defined in prefixItems" % instance
+                )
+            else:
+                for error in validator.descend(
+                        instance,
+                        {'prefixItems': schema['prefixItems']},
+                        path='items__prefixItems',
+                ):
+                    yield error
+    else:
+        if 'prefixItems' in schema:
             for error in validator.descend(
-                item, subschema, path=index, schema_path=index,
+                    instance,
+                    {'prefixItems': schema['prefixItems']},
+                    path='items__prefixItems',
             ):
                 yield error
-    else:
+
+            # Remove evaluated prefixItems indexes
+            del instance[0:len(schema['prefixItems'])]
```

Comment:
We can't ever mutate an instance we're given (it may not even be a mutable type)

Reply:
Fixed